runtime/src/iree/task/pool_test.cc

```diff
@@ -89,4 +89,114 @@ TEST(PoolTest, Trim) {
   iree_task_pool_deinitialize(&pool);
 }
 
+/*
+ Override system allocator to catch failure without building with ASAN.
+ */
```

Comment:
We have a CI build that runs with ASan, so I don't think it should be _necessary_ to add test cases that mimics what ASan does. Do some of the tests fail under ASan without this custom allocator?

cc @GMNGeoffrey

Reply:
Yeah this is pretty nifty but I'm not sure it's worth the complexity to add this to just this one test if we have an asan build in general. I saw the issue you hit was on Arm64. Does this make it fail on other architectures as well?

Reply:
ASAN catches it, so I'll drop the test case. Should have figured ASAN was in the CI...